### PR TITLE
Blocks: switch to new shared, extracted implementation of Tracks

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

**Do not merge, this is waiting on #27937**

This is a follow-up to #27937. Let's start using our shared implementation.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

This will benefit all blocks and more plugins (like Jetpack Social), but I'm building it as part of 1386-gh-dotcom-forge

#### Does this pull request change what data or activity we track or use?

It doesn't change data we track; it will continue to be tracked all the same.

#### Testing instructions:

* Start by enabling upgrade nudges on your test site, it will make it easier to test things: add `add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );` to a functionality plugin.
* Open up the block editor in your browser, open the js console, and send the following: `localStorage.setItem( 'debug', 'dops:analytics*' );`. Refresh the page for good measure.
* In the js console settings, check "Preserve log"
* From that point on, you should see events being logged when you take some actions with Jetpack blocks, like:
    - Add a Paid block like Payments on your free Jetpack site.
    - Publish a post, and click on the Anchor button in the post-publish panel.
    - And other events as seen in this PR.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx [hunk 1]
```diff
@@ -41,8 +43,9 @@ export default createHigherOrderComponent(
 				plan: requiredPlan,
 				blockName: name,
 				context: bannerContext,
+				tracks,
 			} ),
-			[ requiredPlan, name, bannerContext ]
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx:45]: There was no need for `bannerContext` here, the variable cannot change.

**Comment by dereksmart** [comment]
What build steps should be needed? With JN live branch and building plugins/jetpack in docker environment, I'm seeing 

> jetpackConfig is missing in your webpack config file. See @automattic/jetpack-config

In the console, and not seeing events being saved.

**Comment by jeherve** [comment]
> What build steps should be needed?

Just `jetpack build plugins/jetpack`, since this is making changes to the Jetpack plugin.

> jetpackConfig is missing in your webpack config file. See @automattic/jetpack-config

I'm seeing this in `trunk` as well. #27978 should fix it. 

> In the console, and not seeing events being saved.

I do see the events on my end though. Do you not see any, for any of the blocks? Do you see the debug value in your local storage?

File: projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx (hunk 1 above)
Comment (simison) [reply projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx:45]: `react-hooks/exhaustive-deps` might've caught this (and any missing deps elsewhere)

https://reactjs.org/docs/hooks-rules.html
https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks

**Review by dereksmart — APPROVED** [review]

**Comment by roo2** [comment]
I've been seeing an error locally 

`Uncaught TypeError: Cannot destructure property 'pageViewEventName' of '_ref' as it is undefined` coming from the `PaymentsPostPublish` https://github.com/Automattic/jetpack/pull/27940/files#diff-8e5e2d608f1b344e6cefe1cf81a5e8c6cd26e2f62ae597499f65aa4dbb30de64R11

Screen Shot 2022-12-23 at 2 25 17 pm

I'm not sure if I'm missing a build step or anything I'm missing? but it looks like `useAnalytics()` needs an object with `pageViewEventName` which isn't being passed from the `PaymentsPostPublish` block?

**Comment by jeherve** [comment]
> I've been seeing an error locally

Noting that this should now be addressed in #28072.

**Pull request merged**